src/cargo/util/toml/targets.rs

```diff
@@ -106,13 +105,10 @@ fn clean_lib(toml_lib: Option<&TomlLibTarget>,
             let legacy_path = package_root.join("src").join(format!("{}.rs", lib.name()));
             if legacy_path.exists() {
                 {
-                    let short_path = without_prefix(&legacy_path, package_root)
-                        .unwrap_or(&legacy_path);
-
                     warnings.push(format!(
                         "path `{}` was erroneously implicitly accepted for library `{}`,\n\
                          please rename the file to `src/lib.rs` or set lib.path in Cargo.toml",
-                        short_path.display(), lib.name()
+                        legacy_path.display(), lib.name()
                     ));
                 }
```

Comment:
Yep, we don't use `rustfmt` yet, because its default style is still evolving and, at least the last time we tried, it didn't manage to fromat some parts of Cargo prettily. 

IIRC, rustfmt has an option of formating only a certain range of file, which might be useful to format only changed code.